### ISSUE TITLE
test_acceptOutOfFiles is failing on FreeBSD

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2969]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#2969|
|Type|defect|
|Created|2007-12-28 17:27:04Z|

The reason is kind of weird: connect doesn't fail with EINPROGRESS, but succeeds. The test should probably be modified to take care of that.

Attachments:

* [2969.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/df9/df93151d1f40e8349efcabc94af789149f2fe2d5/0b1adfca8a99dc556cdf339c1f36a2053065c2cd.diff) (814 bytes) - added by therve on 2008-02-01 09:41:10Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2969 2969
type__defect defect
reporter__therve therve
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1198862824000000 1198862824000000
changetime__1201886510000000 1201886510000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve removed owner|
|-|-|

Green buildbot ftw.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @therve|
|-|-|

Seems reasonable.  The test doesn't really care what happens to the connect attempt, as long as it tricks accept() into trying to allocate a new file descriptor.  If FreeBSD thinks the connection succeeded, that's fine.  Please apply.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set status to `closed`|
|-|-|

(In [22438]) Fix test_acceptOutOfFiles under FreeBSD: connect (generally) doesn't raise
EINPROGRESS but succeeds immediatly.

Author: therve
Reviewer: exarkun
Fixes [#2969](https://github.com/twisted/twisted/issues/2969)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
